### PR TITLE
fix: Grouper admin raised AttributeError when used outside the admin views

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

This PR adds fallback methods to `GrouperModelAdmin` to allow for fallback methods if not all grouping fields are set. This allows to use `GrouperModelAdmin` outside the admin views, e.g., as a target for the `AutocompleteSelect` widget.

It makes pages searchable by adding search fields. Also, the page's __str__ method is updated for search fields: This way pages are consistently represented in the admin by (`str(page)`) in the form of "My page title (/path/to/page)" in the current or a fallback language.

Finally, a type hint error is fixed.

## Related resources

* #...
* #...

## Checklist

* [x] I have opened this pull request against ``develop-4``
* [x] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined the channel #pr-reviews on our [Discord Server](https://discord-pr-review-channel.django-cms.org) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: cms/admin/pageadmin.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ class PageAdmin(admin.ModelAdmin):
     copy_form = CopyPageForm
     move_form = MovePageForm
     inlines = PERMISSION_ADMIN_INLINES
+    search_fields = ('=id', 'urls__slug', 'pagecontent_set__title', 'reverse_id')
```
Comment (vinitkumar) [inline cms/admin/pageadmin.py:107]: I believe this `=id` is a typo?

**Review by vinitkumar — CHANGES_REQUESTED** [review]

File: cms/admin/pageadmin.py (hunk 1 above)
Comment (marksweb) [reply cms/admin/pageadmin.py:107]: No that's OK.

That makes it an `iexact` search on ID

See examples in the docs
https://docs.djangoproject.com/en/5.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.search_fields

File: cms/models/pagemodel.py [hunk 2]
```diff
@@ -728,7 +728,7 @@ def set_translations_cache(self):
             self.page_content_cache.setdefault(translation.language, translation)
 
     def set_admin_content_cache(self):
-        self.admin_conent_cache = AdminCacheDict()
+        self.admin_content_cache = AdminCacheDict()
```
Comment (marksweb) [inline cms/models/pagemodel.py:731]: I do like to see typos fixed 😁 👏

**Review by marksweb — APPROVED** [review]

File: cms/admin/pageadmin.py (hunk 1 above)
Comment (vinitkumar) [reply cms/admin/pageadmin.py:107]: Ah, right!

**Review by vinitkumar — APPROVED** [review]

File: cms/admin/pageadmin.py (hunk 1 above)
Comment (fsbraun) [reply cms/admin/pageadmin.py:107]: I've translated the search fields from the page content admin for consistency. :-)

**Pull request merged**